src/common/mail.service.ts

```diff
@@ -1,28 +1,25 @@
-import { MailerService } from "@nestjs-modules/mailer";
+import { ISendMailOptions, MailerService } from "@nestjs-modules/mailer";
 import { Injectable, Logger } from "@nestjs/common";
+import { SentMessageInfo } from "nodemailer";
 
+/**
+ * Service for sending emails using nodemailer.
+ *
+ * Use this rather than MailerService directly to allow configuration in AppModule
+ */
 @Injectable()
 export class MailService {
   constructor(private readonly mailerService: MailerService) {}
 
-  async sendMail(
-    to: string,
-    cc: string,
-    subject: string,
-    mailText: string | null,
-    html: string | null = null,
-  ) {
+  async sendMail(options: ISendMailOptions): Promise<SentMessageInfo> {
```

Comment:
I see that we have `nodemailer` package installed separately, and it is used only as a type in this file. However, `@nestjs-modules/mailer` already has nodemailer as part of the dependency package. 
I'd suggest to remove `nodemailer` from the package.json if we have no intention to use this package separately and rely on dependency nodemailer from @nestjs-modules/mailer exclusively

Reply:
Good suggestion. I think `reflect-metadata` is also used only indirectly, but I will save removing that for a different PR.